src/Utils.cpp

```diff
@@ -44,6 +44,37 @@ std::optional<std::string> getAncestorPath(const std::string& path, const std::s
     return std::nullopt;
 }
 
+size_t pathDistance(const std::string& path1, const std::string& path2)
+{
+    std::filesystem::path p1(path1);
+    std::filesystem::path p2(path2);
+
+    auto distance = std::filesystem::relative(p1, p2);
+    return std::distance(distance.begin(), distance.end()) - 1;
+}
+
+std::string convertToScriptPath(const std::string& path)
+{
+    std::filesystem::path p(path);
+    std::string output = "";
+    for (auto& it = p.begin(); it != p.end(); ++it)
+    {
+        auto str = it->string();
+        if (str.find(" ") != std::string::npos)
+            output += "[\"" + str + "\"]";
+        else if (str == ".")
+            output += "script";
+        else if (str == "..")
+            output += "script.Parent";
```

Comment:
I don't think this is completely right: we should only do `script` and `script.Parent` if this is at the start of the path. If its in the middle, it should just be empty or `.Parent`. (Not sure if that case would ever be hit in practice though)